connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -53,36 +56,67 @@
 
     private Map<String, String> config;
     private State state;
+    private OffsetStorageReader offsetStorageReader;
 
     public WorkerConnector(String connName,
                            Connector connector,
                            ConnectorContext ctx,
-                           ConnectorStatus.Listener statusListener) {
+                           ConnectorStatus.Listener statusListener,
+                           OffsetStorageReader offsetStorageReader) {
         this.connName = connName;
         this.ctx = ctx;
         this.connector = connector;
         this.statusListener = statusListener;
         this.state = State.INIT;
+        this.offsetStorageReader = offsetStorageReader;
     }
 
     public void initialize(ConnectorConfig connectorConfig) {
         try {
             this.config = connectorConfig.originalsStrings();
             log.debug("{} Initializing connector {} with config {}", this, connName, config);
-
-            connector.initialize(new ConnectorContext() {
+            final ConnectorContext delegateCtx = new ConnectorContext() {
                 @Override
                 public void requestTaskReconfiguration() {
-                    ctx.requestTaskReconfiguration();
+                    WorkerConnector.this.ctx.requestTaskReconfiguration();
                 }
 
                 @Override
                 public void raiseError(Exception e) {
                     log.error("{} Connector raised an error", this, e);
                     onFailure(e);
-                    ctx.raiseError(e);
+                    WorkerConnector.this.ctx.raiseError(e);
                 }
-            });
+            };
+            if (isSinkConnector())
+                connector.initialize(new SinkConnectorContext() {
+                    @Override
+                    public void requestTaskReconfiguration() {
+                        delegateCtx.requestTaskReconfiguration();
+                    }
+
+                    @Override
+                    public void raiseError(Exception e) {
+                        delegateCtx.requestTaskReconfiguration();
```

Comment:
Is it possible to trigger infinite loop:

raiseError -> reconfig -> raiseError -> reconfig ...